syft/pkg/cataloger/cataloger.go

```diff
@@ -32,22 +32,22 @@ type Cataloger interface {
 }
 
 // ImageCatalogers returns a slice of locally implemented catalogers that are fit for detecting installations of packages.
-func ImageCatalogers() []Cataloger {
+func ImageCatalogers(cfg Config) []Cataloger {
 	return []Cataloger{
 		ruby.NewGemSpecCataloger(),
 		python.NewPythonPackageCataloger(),
 		php.NewPHPComposerInstalledCataloger(),
 		javascript.NewJavascriptPackageCataloger(),
 		deb.NewDpkgdbCataloger(),
 		rpmdb.NewRpmdbCataloger(),
-		java.NewJavaCataloger(),
+		java.NewJavaCataloger(cfg.Java()),
```

Comment:
would a better pattern maybe be passing the entire `cfg` to each cataloger that needs it as some might share the same configuration

Reply:
I toyed with that idea but leaned towards the direction that focused the concerns for each cataloger. Technically each cataloger should be able to describe exactly what it needs in a small configuration which callers can use (in this case, the `java` cataloger package having it's own `Config` description). The downside of the other direction is that the `java` package would be depending on definitions from the `cataloger` package, which would lead to a circular dependency.